File: ゼミ用/05製造/api/src/graph-dev/request/EditGroupRequest.php
<?php

/**
 * グループ編集処理の値を保持する
 */
class EditGroupRequest {
    private $groupId;
    private $groupName;
    private $groupInfo;
    private $groupColor;
    private $userId;
    private $version = 0;
    private $errorMsg = array();

    /**
     * グループ編集処理の値を格納します
     */
    public function __construct() {
    }

    public function getGroupId() {
        return $this->groupId;
    }

    public function setGroupId($groupId) {
        $this->groupId = $groupId;
    }

    public function getGroupName() {
        return $this->groupName;
    }

    public function setGroupName($groupName) {
        $this->groupName = $groupName;
    }

    public function getGroupInfo() {
        return $this->groupInfo;
    }

    public function setGroupInfo($groupInfo) {
        $this->groupInfo = $groupInfo;
    }

    public function getGroupColor() {
        return $this->groupColor;
    }

    public function setGroupColor($groupColor) {
        $this->groupColor = $groupColor;
    }

    public function getUserId() {
        return $this->userId;
    }

    public function setUserId($userId) {
        $this->userId = $userId;
    }

    public function getVersion() {
        return $this->version;
    }

    public function setVersion($version) {
        $this->version = $version;
    }

    public function getErrorMsg() {
        return $this->errorMsg;
    }

    /**
     * バリデーションチェックをします
     */
    public function validation() {
        $validationFlg = false;

        try {
            // 入力項目チェック
            if (empty($this->groupId)) {
                array_push($this->errorMsg, 'エラー ：必須項目が入力されていません');
                $validationFlg = true;
            }

            // グループIDの文字数チェック
            if (strlen($this->groupId) != Constant::GROUP_ID_DIGIT + strlen(Constant::GROUP_ID_STR)) {
                array_push($this->errorMsg, 'エラー ：グループIDは' . Constant::GROUP_ID_DIGIT + strlen(Constant::GROUP_ID_STR) . '文字で入力してください');
                $validationFlg = true;
            }

            // ユーザIDの文字数チェック
            if (strlen($this->userId) != Constant::USER_ID_DIGIT + strlen(Constant::USER_ID_STR)) {
                array_push($this->errorMsg, 'エラー ：ユーザIDは' . Constant::USER_ID_DIGIT + strlen(Constant::USER_ID_STR) . '文字で入力してください');
                $validationFlg = true;
            }

            // グループ名の文字数チェック
            if (!empty($this->groupName)  && 100 < strlen($this->groupName)) {
                array_push($this->errorMsg, 'エラー ：グループ名は100文字以内で入力してください');
                $validationFlg = true;
            }

            // グループ詳細の文字数チェック
            if (!empty($this->groupInfo) && 1200 < strlen($this->groupInfo)) {
                array_push($this->errorMsg, 'エラー ：グループ詳細は1200文字以内で入力してください');
                $validationFlg = true;
            }

            // グループ色の文字数チェック
            if (!empty($this->groupColor) && 100 < strlen($this->groupColor)) {
                array_push($this->errorMsg, 'エラー ：グループ色は100文字以内で入力してください');
                $validationFlg = true;
            }
        } catch (Exception $e) {
            $validationFlg = true;
        }

        return $validationFlg;
    }
}

File: ゼミ用/05製造/eeldev/eelPy/api/request/EditGroupRequest.py
from ..common.Constant import Constant


# /**
# * グループ編集処理の値を保持する
# */
class EditGroupRequest:
    errorMsg = []

    # /**
    # * グループ編集処理の値を格納します
    # */
    def __init__(self, groupId, groupName, groupInfo, groupColor, userId, version):
        self.groupId = groupId
        self.groupName = groupName
        self.groupInfo = groupInfo
        self.groupColor = groupColor
        self.userId = userId
        self.version = version

    def getGroupId(self):
        return self.groupId

    def getGroupName(self):
        return self.groupName

    def getGroupInfo(self):
        return self.groupInfo

    def getGroupColor(self):
        return self.groupColor

    def getUserId(self):
        return self.userId

    def getVersion(self):
        return self.version

    def getErrorMsg(self):
        return self.errorMsg

    # /**
    # * バリデーションチェックをします
    # */
    def validation(self):
        validationFlg = False

        try:
            # 入力項目チェック
            if (len(self.groupId) == 0):
                self.errorMsg.append('エラー ：必須項目が入力されていません')
                validationFlg = True

            # グループIDの文字数チェック
            if (Constant.GROUP_ID_DIGIT + len(Constant.GROUP_ID_STR) < len(self.groupId)):
                self.errorMsg.append(
                    'エラー ：グループIDは' + Constant.GROUP_ID_DIGIT + len(Constant.GROUP_ID_STR) + '文字以内で入力してください')
                validationFlg = True

            # ユーザIDの文字数チェック
            if (Constant.USER_ID_DIGIT + len(Constant.USER_ID_STR) < len(self.userId)):
                self.errorMsg.append(
                    'エラー ：ユーザIDは' + Constant.USER_ID_DIGIT + len(Constant.USER_ID_STR) + '文字以内で入力してください')
                validationFlg = True

            # グループ名の文字数チェック
            if (len(self.groupName) != 0 and 100 < len(self.groupName)):
                self.errorMsg.append('エラー ：グループ名は100文字以内で入力してください')
                validationFlg = True

            # グループ詳細の文字数チェック
            if (len(self.groupInfo) != 0 and 1200 < len(self.groupInfo)):
                self.errorMsg.append('エラー ：グループ詳細は1200文字以内で入力してください')
                validationFlg = True

            # グループ色の文字数チェック
            if (len(self.groupColor) != 0 and 100 < len(self.groupColor)):
                self.errorMsg.append('エラー ：グループ色は100文字以内で入力してください')
                validationFlg = True          

        except Exception as e:
            validationFlg = True

        return validationFlg
